### DISCUSSION TITLE
Finding out the advertised NAT64 prefix on an end device.

### DISCUSSION THREAD

**Discussion by YSLew-tado** [openthread/openthread#8335] [Q&A] [answered]
Hi,

My end device displays a NAT64 prefix with `netdata show`. It is able to reach the public internet when the destination IPv6 address is appended to the prefix.
I was wondering if there is an OT API to find out the advertised NAT64 prefix so that I can utilise it in the firmware instead of CLI.

Thanks

**Comment by jwhui** [comment, marked as answer]
NAT64 prefixes configured on the Thread network are published in Thread network data. Note that there may be one or more NAT64 prefixes configured.

Use [`otNetDataGetNextRoute()`](https://openthread.io/reference/group/api-thread-general#otnetdatagetnextroute) to iterate through the routes published in Thread network data. The [`otExternalRouteConfig`](https://openthread.io/reference/struct/ot-external-route-config) structure indicates whether the route is a NAT64 prefix.

The CLI implementation of `netdata show` provides an example of how to use these APIs: https://github.com/openthread/openthread/blob/main/src/cli/cli_network_data.cpp#L503

**Reply by YSLew-tado** [reply]
That helped.

Thanks for pointing me to the right place!

**Comment by Zepperoni** [comment]
@YSLew-tado Also trying to figure out the NAT64 prefix. Could you provide me with an example

**Reply by jwhui** [reply]
The [otNat64SynthesizeIp6Address()](https://openthread.io/reference/group/api-nat64#otnat64synthesizeip6address) API will synthesize an IPv6 address from an IPv4 address using the advertised NAT64 prefix.

**Reply by Zepperoni** [reply]
Perfect, that helped me.
Thanks
